### PR TITLE
 OZ-560: Rename project,Added E2E tests and registration Opt In

### PR DESCRIPTION
**Pull request by enyachoke**
This is a PR for the work in https://mekomsolutions.atlassian.net/browse/OZ-560. It introduces a number of changes to the original project;

- Container registration is now opt-in
- Upgrade a number of libraries
- Adds E2E Tests

### CODE REVIEW

File: README.md
Comment (rbuisson) [inline README.md]: Should we keep this Docker image name?

File: README.md
Comment (rbuisson) [inline README.md]: Are these going to work?

File: README.md
Comment (rbuisson) [inline README.md]: Let's have the container named `container-to-consul` if possible.

File: e2e/container2consul.spec.ts
Comment (kdaud) [inline e2e/container2consul.spec.ts]:
```suggestion
  await new Promise(r => setTimeout(r, 5000));
```

File: README.md
Comment (rbuisson) [inline README.md]: Isn't this based on https://github.com/benoitvidis/con-tainer2sul instead?

File: README.md
Comment (rbuisson) [inline README.md]:
```suggestion
Container-to-Consul automatically registers and deregisters any Docker container as HashiCorp Consul nodes.
```

**Review by kdaud — CHANGES_REQUESTED** [review]

File: README.md
Comment (rbuisson) [reply README.md]: If so, it needs more emphasis.
What about:
> This project is largely based on the original work done in [con-tainer2sul](https://github.com/benoitvidis/con-tainer2sul) by [Benoît Vidis](https://github.com/benoitvidis/). While this fork includes significant changes and improvements, we would like to acknowledge the foundation provided by the original project, without which this version would not have been possible.

> Please note that some files, variables, or other elements might still reference the original project name. These remnants will be updated over time as the project evolves, but their presence reflects the project's roots as `con-tainer2sul`.

File: README.md
Comment (rbuisson) [inline README.md:1]: @enyachoke  the doc specifies multiple commands to be run, with different option. Can those options be used simultaneously? eg, using `consul.service=myservice` and `consul.tags=xyz`

**Comment by rbuisson** [comment]
@enyachoke  could you provide some instructions to test?
The Docker image `mekomsolutions/container2sul` does not seem to be available so how do I build it?

**Comment by enyachoke** [comment]
> @enyachoke could you provide some instructions to test? The Docker image `mekomsolutions/container2sul` does not seem to be available so how do I build it?

@rbuisson This will be available after we merge see
https://github.com/mekomsolutions/container-to-consul/blob/52f007b0788d2fecc29abd6c7b6d7fe8cce04fe6/.github/workflows/build.yml#L13-L19

File: e2e/container2consul.spec.ts
Comment (enyachoke) [reply e2e/container2consul.spec.ts]: @kdaud  is this based on your tests locally?

File: README.md
Comment (enyachoke) [reply README.md:1]: Yes they can as used in the tests https://github.com/mekomsolutions/container-to-consul/blob/52f007b0788d2fecc29abd6c7b6d7fe8cce04fe6/e2e/docker/test-compose.yaml#L4-L10

File: e2e/container2consul.spec.ts
Comment (kdaud) [reply e2e/container2consul.spec.ts]: Setting 3000 milliseconds equates to 3 seconds, and the comment indicates that a pause of 5 minutes is required to allow the Nginx test containers to start.

**Comment by rbuisson** [comment]
Yes I know. So what are the instructions?
docker build first? (what exact command). And then what?

File: README.md
Comment (rbuisson) [reply README.md:1]: In that case, the structure of the doc is misleading.
Why would someone want to run those commands one by one like that?

I suppose it is better to simply document what the options can do, without an example `docker run --rm ...` which in fact will probably never be used.

Comment (enyachoke) [reply README.md:1]: > In that case, the structure of the doc is misleading. Why would someone want to run those commands one by one like that?
> 
> I suppose it is better to simply document what the options can do, without an example `docker run --rm ...` which in fact will probably never be used.

The commands are used to explain the effect of the option and are not necessarily complete usage examples.

File: .github/workflows/build.yml
Comment (rbuisson) [inline .github/workflows/build.yml]:
```suggestion
  docker-publish:
```

File: package.json
Comment (rbuisson) [inline package.json:13]: Could you add a new key for contributors:

`
  "contributors": [
    "Mekom Solutions (https://mekomsolutions.com/)"
  ]
`

File: README.md
Comment (rbuisson) [reply README.md]: Did you remove these lines?

File: README.md
Comment (rbuisson) [inline README.md]:
```suggestion
docker run --name nginx -d \
 -l consul.register=true \
  -l consul.ip=10.0.0.1 \
   -l consul.service=nginx \
    -l consul.tags=www,api \
     -l consul.port=80 \
      -l consul.kv.foo=bar \
       -l consul.kv.bar=baz \
        nginx
```

File: README.md
Comment (rbuisson) [inline README.md]: Is removing the container necessary to deregister? I don't think so, right?
I suggest to then remove the `rm` command to put it a bit after, as a note or something.

**Review by rbuisson — CHANGES_REQUESTED** [review]
I've pushed a change on the README to add some screenshots.
I've left some comments as well.

But it works!
Thanks, that's an awesome tool!

**Review by kdaud — APPROVED** [review]
LGTM

@rbuisson needs to do a final review of the PR.

**Review by rbuisson — APPROVED** [review]

**Pull request merged**
